src/main/scala/org/zalando/spearheads/innkeeper/api/JsonDtos.scala

```diff
@@ -22,10 +22,29 @@ object RouteName {
 
 case class TeamName(name: String) extends AnyRef
 
+case class UserName(name: String) extends AnyRef {
+
+  def apply(name: Option[String]) = name match {
+    case Some(name) => UserName(name)
+    case _          => throw InvalidUsereNameException
+  }
+}
+
+object UserName {
+  def apply(name: Option[String]): UserName = name match {
```

Comment:
That seems like a weird syntax/API to me. Having an apply with an Option that throws on none... Why not just have an apply with a pure string and let the caller of that thing handle the case of empty Options? This way, I always have to wrap a String into an Option to create a Username? seems weird